virttest/qemu_devices/qdevices.py

```diff
@@ -1900,6 +1907,15 @@ def get_free_root_port(self):
 
         return None
 
+    def hotplug(self, device):
+        """
+        Hotplug device, plug into a free root port
+        :param device: The QDevice object
+        """
+        root_port = self.get_free_root_port()
+        device.set_param("bus", root_port)
+        device.parent_bus += ({"busid": root_port},)
```

Comment:
Yep, I do like this. It should allow sufficient flexibility for now. In the future we might need to add some arguments, but for now it's more than enough...